### PR TITLE
[IMP] udes_stock: Override _set_u_result_parent_package_id

### PR DESCRIPTION
**Pull request by unirho**
Overriden _set_u_result_parent_package_id to not add parent package
if the u_target_storage_format is a package/pallet of packages,
this facilitates correctly storing as packages in pick face from
returns putaway/internal transfer.

User-story: 9570
Signed-off-by: Robert Smith <robert.smith@unipart.io>

### CODE REVIEW

File: addons/udes_stock/models/stock_picking.py [hunk 1]
```diff
@@ -1244,17 +1244,23 @@ def unlink_empty(self):
...
-        if package_move_lines:
-            package_move_lines.write({'u_result_parent_package_id': False})
+
+    def _set_u_result_parent_package_id(self):
+        """ Override function to avoid setting result_package_id when user
+        the storage format is package/pallet of products"""
+        package_pickings = self \
+            .filtered(
+                lambda p: p.picking_type_id.u_target_storage_format in
+                ('package' or 'pallet_packages')
```
Comment (ws-mps) [inline addons/udes_stock/models/stock_picking.py]: ('package', 'pallet_packages')

Comment (ws-mps) [reply addons/udes_stock/models/stock_picking.py]: In fact, this should be pallet products not pallet packages

Comment (ws-mps) [reply addons/udes_stock/models/stock_picking.py]: Also we don't want to set it for products. So probably do it the other way around:
`pallet_package_pickings = XXX`
and use it when calling super.

File: addons/udes_stock/models/stock_picking.py [hunk 2]
```diff
@@ -1244,17 +1244,22 @@ def unlink_empty(self):
...
         super(StockPicking, pickings)._check_entire_pack()
-        package_move_lines = pickings \
-            .filtered(lambda p: p.picking_type_id.u_target_storage_format == 'package') \
-            .mapped("move_line_ids")
-        if package_move_lines:
-            package_move_lines.write({'u_result_parent_package_id': False})
+
+    def _set_u_result_parent_package_id(self):
+        """ Override function to avoid setting result_package_id when user
+        the storage format is package/pallet of products"""
```
Comment (ws-mps) [inline addons/udes_stock/models/stock_picking.py]: products/package/pallet of products

File: addons/udes_stock/tests/test_picking.py [hunk 3]
```diff
@@ -16,6 +16,30 @@ def setUpClass(cls):
                                               confirm=True)
         cls.SudoPicking = Picking.sudo(cls.inbound_user)
         cls.test_picking = cls.test_picking.sudo(cls.inbound_user)
+        cls.tangerine_lot = cls.create_lot(cls.tangerine.id, 1)
+
+    def picksByStorageType(self, storage_type=False):
```
Comment (nocturnalastro) [inline addons/udes_stock/tests/test_picking.py]: snakecase

File: addons/udes_stock/tests/test_picking.py [hunk 4]
```diff
@@ -16,6 +16,30 @@ def setUpClass(cls):
...
+        """
+            Generate and return picks and pallets for different storage formats
+        """
+        Package = self.env['stock.quant.package']
+        self.picking_type_in.u_target_storage_format = storage_type
```
Comment (nocturnalastro) [inline addons/udes_stock/tests/test_picking.py]: do this in the test then you don't need to pass the parameter

File: addons/udes_stock/tests/test_picking.py [hunk 5]
```diff
@@ -16,6 +16,30 @@ def setUpClass(cls):
...
+        mummy_pallet = Package.get_package('mummy_pallet', create=True)
```
Comment (nocturnalastro) [inline addons/udes_stock/tests/test_picking.py]: Move these into setup.

File: addons/udes_stock/tests/test_picking.py [hunk 6]
```diff
@@ -102,3 +126,16 @@ def test08_related_pickings(self):
...
+    def test09_only_pallets_of_packages_have_parent_package(self):
+        """
+            Test that only pallets of packages have a parent package added by
+           _check_entire_pack/_set_u_result_parent_package_id
+        """
+
+        pallet, pick = self.picksByStorageType("pallet_packages")
+        self.assertEqual(pallet, pick.move_line_ids.u_result_parent_package_id)
+
+        for storage_type in ("product", "pallet_products", "package"):
```
Comment (nocturnalastro) [inline addons/udes_stock/tests/test_picking.py]: these should probably be different tests.

**Pull request merged**